### PR TITLE
Send alert to Slack on Workflow Fail

### PR DESCRIPTION
**Pull request by bitsofsteve**
## Description

This PR aims to send a notification on slack when a workflow FAILS on CI, this will enable easy monitoring for the CI.

### CODE REVIEW

File: .github/workflows/stack-integration_tests.yml [hunk 1]
```diff
@@ -653,3 +653,19 @@ jobs:
...
+
+      - name: Job Report Status
+        if: always()
+        uses: ravsamhq/notify-slack-action@v1
+        with:
+          status: ${{ job.status }}
+          notify_when: 'failure'
+          notification_title: '{workflow} has {status_message}'
+          message_format: '{emoji} *{workflow}* {status_message} in <{repo_url}|{repo}>'
+          footer: 'Linked to Repo <{workflow_url}|{repo_url}>'
```
Comment (shubham3121) [inline .github/workflows/stack-integration_tests.yml:665]: Maybe we also attach the run_url to either message or footer

File: .github/workflows/stack-integration_tests.yml [hunk 2]
```diff
@@ -653,3 +653,19 @@ jobs:
...
           docker rm `docker ps -aq` --force || true
+
+      - name: Job Report Status
+        if: always()
+        uses: ravsamhq/notify-slack-action@v1
+        with:
+          status: ${{ job.status }}
+          notify_when: 'failure'
+          notification_title: '{workflow} has {status_message}'
+          message_format: '{emoji} *{workflow}* {status_message} in <{repo_url}|{repo}>'
```
Comment (shubham3121) [inline .github/workflows/stack-integration_tests.yml:664]: Maybe we can add the branch_url so we have link to the branch itself.

**Review by shubham3121 — CHANGES_REQUESTED** [review]
Left some comments. Otherwise looks great !! 👏🏾

**Comment by rasswanth-s** [comment]
@bitsofsteve  would this  PR functionality be part of the 0.7 release or future release?

**Comment by madhavajay** [comment]
@bitsofsteve Looks like a conflict, can you fix it and make the requested changes so we can get this into production?

**Comment by bitsofsteve** [comment]
> @bitsofsteve Looks like a conflict, can you fix it and make the requested changes so we can get this into production?

Sure @madhavajay !

File: .github/workflows/stack-integration_tests.yml (hunk 1 above)
Comment (bitsofsteve) [reply .github/workflows/stack-integration_tests.yml]: This is done..

File: .github/workflows/stack-integration_tests.yml (hunk 2 above)
Comment (bitsofsteve) [reply .github/workflows/stack-integration_tests.yml]: @shubham3121 the {branch_url} doesn't seem to generate a correct context  URL After all, there are workarounds but will take some hacking. For the moment, I have manually constructed and added the PR URL instead at the footer and this should work for now.

**Comment by bitsofsteve** [comment]
> @bitsofsteve Looks like a conflict, can you fix it and make the requested changes so we can get this into production?

@rasswanth-s @madhavajay this is ready for merge.

File: .github/workflows/stack-integration_tests.yml (hunk 2 above)
Comment (shubham3121) [reply .github/workflows/stack-integration_tests.yml]: Great !! 👍🏾

**Review by madhavajay — APPROVED** [review]
Great work @bitsofsteve lets see how she flies! :)

**Pull request merged**